server/apis/v1_1/handler.go

```diff
@@ -347,31 +372,38 @@ func (h *handler) UpdateInterStepBufferService(c *gin.Context) {
 		c.JSON(http.StatusOK, NewNumaflowAPIResponse(&errMsg, nil))
 		return
 	}
-	var requestBody dfv1.InterStepBufferServiceSpec
-	err = json.NewDecoder(c.Request.Body).Decode(&requestBody)
+	//var requestBody dfv1.InterStepBufferServiceSpec
+	//err = json.NewDecoder(c.Request.Body).Decode(&requestBody)
+	requestBody, err := parseSpecFromReq(c, SpecTypeISB)
 	if err != nil {
 		errMsg := fmt.Sprintf("Failed to update the interstep buffer service: namespace %q isb-services %q: %v", c.Param("namespace"), c.Param("isb-services"), err.Error())
 		c.JSON(http.StatusOK, NewNumaflowAPIResponse(&errMsg, nil))
 		return
 	}
-
-	if requestBody.Redis != nil {
+	var updatedSpec = requestBody.(*dfv1.InterStepBufferService)
+	isValid := validateISBSpec(h, isbSVC, updatedSpec, ValidTypeUpdate)
```

Comment:
I think for the isbsvc validation we can just keep this function and remove the other condition checks below... what do you think?